examples/notebooks/beam-ml/rag_usecase/opensearch_connector.py

```diff
@@ -0,0 +1,383 @@
+    to connect to Opensearch DB
+    """
+    
+    def __init__(self, 
+                 host:str, 
+                 port:int, 
+                 username:str,
+                 password:str,
+                 batch_size:int=100,
+                 embedded_columns:list=[]):
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.batch_size = batch_size
+        self.embedded_columns = embedded_columns
+
+        self.batch_counter = 0
+        self.batch = list()
+
+    def finish_bundle(self):
+        self._flush()
+
+    def process(self, element, *args, **kwargs):
+        self.batch.append(element)
+        self.batch_counter += 1
+        if self.batch_counter >= self.batch_size:
+            self._flush()
+
+    def _flush(self):
+        if self.batch_counter == 0:
+            return
+
+        with _InsertEmbeddingInOpenSearchSink(self.host, self.port, self.username, self.password, self.embedded_columns) as sink:
+
+            sink.write(self.batch)
+
+            self.batch_counter = 0
+            self.batch = list()
+
+
+class _InsertEmbeddingInOpenSearchSink(object):
+    """Class where we create Opensearch client 
+    and write text embedding  in Opensearch DB
+    """
+    
+    def __init__(self, host:str, 
+                 port:int, 
+                 username:str,
+                 password:str,
+                 embedded_columns:list=[]):
+        self.host = host
+        self.port = port
+        self.username = username
+        self.password = password
+        self.embedded_columns = embedded_columns
+        self.client = None
+
+    def _create_client(self):
+        if self.client is None:
+            http_auth = [self.username, self.password]
+            self.client = OpenSearch(hosts = [f'{self.host}:{self.port}'],
+                                    http_auth = http_auth,
+                                    verify_certs = False
+        )
+
+        
+    def write(self, elements):
+        self._create_client()
+        documents = []
+        for element in elements:
+            print(f'Insert Embeddings in opensearch DB {element}')
+            doc_update = {
+                "url" : element["url"],
+                "section_id" : element["section_id"]
+            }
+            
+            for k,v in element.items():
+                if k in self.embedded_columns:
+                    doc_update[f"{k}_vector"] = v
+                        
+            documents.extend([{
+                            "update": {
+                            "_index": "embeddings-index", 
+                            "_id" : str(element["id"]),
+                            }
+                        }, { 
+                            "doc": doc_update
+                        }])
+                    
+        
+        print(f'Inside insert Doc in DB{documents}')
+        self.client.bulk(documents)
```

Comment:
Does this call need any sort of error checking/handling?

Reply:
Added the error handling